pkg/mount/dag_steps.go

```diff
@@ -384,10 +385,30 @@ func (s *State) WriteSentinelDagStep(g *herd.Graph, deps ...string) error {
 			// Lets add a uki sentinel as well!
 			cmdline, _ := os.ReadFile(internalUtils.GetHostProcCmdline())
 			if strings.Contains(string(cmdline), "rd.immucore.uki") {
+				// Generic sentinel for uki mode
+				// TODO: Drop this from the layout file and use the ones below to check our uki mode
 				err = os.WriteFile("/run/cos/uki_mode", []byte("1"), os.ModePerm)
+				internalUtils.Log.Info().Str("to", "uki_mode").Msg("Setting sentinel file")
 				if err != nil {
 					return err
 				}
+				// Specific one
+				err = internalUtils.CheckEfiPartUUID()
+				if err != nil {
+					internalUtils.Log.Info().Str("to", "uki_install_mode").Msg("Setting sentinel file")
+					err2 := os.WriteFile("/run/cos/uki_install_mode", []byte("1"), os.ModePerm)
+
+					if err2 != nil {
+						return err2
+					}
+					return nil
+				} else {
+					internalUtils.Log.Info().Str("to", "uki_boot_mode").Msg("Setting sentinel file")
+					err = os.WriteFile("/run/cos/uki_boot_mode", []byte("1"), os.ModePerm)
+					if err != nil {
+						return err
+					}
```

Comment:
I'm a bit confused why this is different than what happens in the above `if` section, feels to me like the error check is not needed because in both cases it does the same ... but I'm a bit jetlagged still so maybe I'm missreading it haha